core/sdk-core/src/test/java/software/amazon/awssdk/core/internal/retry/RateLimitingTokenBucketTest.java

```diff
@@ -96,6 +99,62 @@ public void tryAcquireCapacity_capacitySufficient_returns0() {
         tb.setCurrentCapacity(5.0);
 
         assertThat(tb.tryAcquireCapacity(5.0)).isZero();
+        assertThat(tb.getCurrentCapacity()).isZero();
+    }
+
+    @Test
+    public void tryAcquireCapacity_amountGreaterThanCapacity_returnsNonZero() {
+        RateLimitingTokenBucket tb = new RateLimitingTokenBucket();
+        tb.setCurrentCapacity(5.0);
+
+        assertThat(tb.tryAcquireCapacity(8.0)).isEqualTo(3);
+        assertThat(tb.getCurrentCapacity()).isNegative();
+        assertThat(tb.getCurrentCapacity()).isEqualTo(-3);
+    }
+
+
+    @Test
+    public void acquire_amountGreaterThanNonZeroCapacity_sleepsForRequiredTime() {
+        RateLimitingTokenBucket tb = Mockito.spy(new RateLimitingTokenBucket());
+
+        // stub out refill() so we have control over the capacity
+        Mockito.doAnswer(invocationOnMock -> null).when(tb).refill();
+
+        tb.setFillRate(1.0);
+        tb.setCurrentCapacity(1.0);
+        tb.enable();
+
+        // 1 token to wait for at a rate of 1 per second should sleep for approx 1s
+        long a = System.nanoTime();
+        boolean acquired = tb.acquire(3.0);
+        long elapsed = System.nanoTime() - a;
```

Comment:
Can you just mock the `RateLimitingTokenBucket.sleep()` method so we don't actually need to wait 3 seconds?